pretty_midi/pretty_midi.py

```diff
@@ -382,6 +397,16 @@ def __get_instrument(program, channel, track, create_new):
                         program, event.channel, track_idx, 0)
                     # Add the control change event
                     instrument.control_changes.append(control_change)
+            # Find instument with maximum lyrics matches
+            channel = np.argmax(lyrics_matches)
```

Comment:
Many MIDI files will not have any lyrics instruments (i.e. lyrics_matches will be zero for all instruments) and some MIDI files will have more than one. Why not consider an instrument a vocal instrument if there is at least one lyric event?

Reply:
It is not that simple, since a lyric event may coincide with multiple instruments, i.e. a guitar strum and a vocal note simultaneously. This is the method MuseScore uses to discover vocal instruments, for example. It could fail in the event of multiple singers, obviously.